components/eam/src/chemistry/mozart/mo_drydep.F90

```diff
@@ -2523,723 +520,684 @@ subroutine drydep_xactive( ncdate, sfc_temp, pressure_sfc,  &
+          ustarb = cvarb * va(icol)/(1._r8 + 4.7_r8*ribn(icol))
+       endif
+       uustar(icol) = va(icol)*ustarb
+    enddo
+
+  end subroutine calculate_uustar
+
+
+  subroutine calculate_ustar(ncol, beglt, endlt, index_season, fr_lnduse, unstable, zl, uustar, ribn, & ! in
+                             ustar, cvar, bycp)                                                         ! out
+
+    use seq_drydep_mod, only: z0
+
+    ! input
+    integer, intent(in) :: ncol
+    integer, intent(in) :: beglt
+    integer, intent(in) :: endlt
+    integer, intent(in) :: index_season(ncol,n_land_type)
+    logical, intent(in) :: fr_lnduse(ncol, n_land_type)
+    logical, intent(in) :: unstable(ncol)
+    real(r8), intent(in) :: zl(ncol)                      ! height of lowest level [m]
+    real(r8), intent(in) :: uustar(ncol)                  ! u*ustar (assumed constant over grid) [m^2/s^2]
+    real(r8), intent(in) :: ribn(ncol)                    ! richardson number [unitless]
+
+    ! output
+    real(r8), intent(out) :: ustar(ncol,n_land_type)      ! friction velocity [m/s]
+    real(r8), intent(out) :: cvar(ncol, n_land_type)      ! height parameter
+    real(r8), intent(out) :: bycp(ncol, n_land_type)      ! buoyancy parameter for unstable conditions
+
+    ! local variables
+    integer :: icol, lt
+    real(r8) :: z0water ! revised z0 over water
+    real(r8), parameter    :: diffk       = 1.461e-5_r8
 
     !-------------------------------------------------------------------------------------
     ! calculate the friction velocity for each land type u_i=uustar/u*_i
     !-------------------------------------------------------------------------------------
     do lt = beglt,endlt
-       do i = 1,ncol
-          if( fr_lnduse(i,lt) ) then
-             if( unstable(i) ) then
-                cvar(i,lt)  = vonkar/log( z(i)/z0(index_season(i,lt),lt) )
-                b(i,lt)     = 9.4_r8*(cvar(i,lt)**2)* sqrt( abs(ribn(i))*z(i)/z0(index_season(i,lt),lt) )
-                ustar(i,lt) = sqrt( cvar(i,lt)*uustar(i)*sqrt( 1._r8 - (9.4_r8*ribn(i)/(1._r8 + 7.4_r8*b(i,lt))) ) )
+       do icol = 1,ncol
+          if( fr_lnduse(icol,lt) ) then
+             if( unstable(icol) ) then
+                cvar(icol,lt)  = karman/log( zl(icol)/z0(index_season(icol,lt),lt) )
+                bycp(icol,lt)  = 9.4_r8*(cvar(icol,lt)**2)* sqrt( abs(ribn(icol))*zl(icol)/z0(index_season(icol,lt),lt) )
+                ustar(icol,lt) = sqrt( cvar(icol,lt)*uustar(icol)*sqrt( 1._r8 - (9.4_r8*ribn(icol)/(1._r8 + 7.4_r8*bycp(icol,lt))) ) )
              else
-                cvar(i,lt)  = vonkar/log( z(i)/z0(index_season(i,lt),lt) )
-                ustar(i,lt) = sqrt( cvar(i,lt)*uustar(i)/(1._r8 + 4.7_r8*ribn(i)) )
-             end if
-          end if
-       end do
-    end do
+                cvar(icol,lt)  = karman/log( zl(icol)/z0(index_season(icol,lt),lt) )
+                ustar(icol,lt) = sqrt( cvar(icol,lt)*uustar(icol)/(1._r8 + 4.7_r8*ribn(icol)) )
```

Comment:
looks that if you define `z0water = z0(index_season(icol,lt),lt)`, this loop will be the same as the loop below and may be separate out as a subroutine.

Reply:
made another subroutine for ustar over water

Reply:
Is this change committed and pushed? I don't see the change.

Reply:
see below lines 736 to 778, I made another subroutine `calculate_ustar_over_water`. Note that the two do loops in `calculate_ustar` and `calculate_ustart_over_water` is not the same. The below one only correct when `lt = lt_for_water`